src/sentry/utils/auth.py

```diff
@@ -269,6 +269,12 @@ def log_auth_failure(request, username=None):
     })
 
 
+def has_user_registration():
+    from sentry import features, options
+
+    return features.has('auth:register') and options.get('auth.allow-registration')
```

Comment:
does this not mean that people that upgrade will have user registration disabled?